Code/Framework/AzToolsFramework/AzToolsFramework/ActionManager/Action/ActionManager.h

```diff
@@ -62,18 +65,30 @@ namespace AzToolsFramework
         ActionManagerOperationResult RegisterActionUpdater(const AZStd::string& actionUpdaterIdentifier) override;
         ActionManagerOperationResult AddActionToUpdater(const AZStd::string& actionUpdaterIdentifier, const AZStd::string& actionIdentifier) override;
         ActionManagerOperationResult TriggerActionUpdater(const AZStd::string& actionUpdaterIdentifier) override;
+        ActionManagerOperationResult RegisterWidgetAction(
+            const AZStd::string& widgetActionIdentifier,
+            const WidgetActionProperties& properties,
+            AZStd::function<QWidget*()> generator
+        ) override;
+        bool IsWidgetActionRegistered(const AZStd::string& widgetActionIdentifier) const override;
+        ActionManagerGetterResult GetWidgetActionName(const AZStd::string& widgetActionIdentifier) override;
+        ActionManagerOperationResult SetWidgetActionName(const AZStd::string& widgetActionIdentifier, const AZStd::string& name) override;
+        ActionManagerGetterResult GetWidgetActionCategory(const AZStd::string& widgetActionIdentifier) override;
+        ActionManagerOperationResult SetWidgetActionCategory(const AZStd::string& widgetActionIdentifier, const AZStd::string& category) override;
 
         // ActionManagerInternalInterface overrides ...
         QAction* GetAction(const AZStd::string& actionIdentifier) override;
         const QAction* GetActionConst(const AZStd::string& actionIdentifier) const override;
         bool GetHideFromMenusWhenDisabled(const AZStd::string& actionIdentifier) const override;
         bool GetHideFromToolBarsWhenDisabled(const AZStd::string& actionIdentifier) const override;
+        QWidget* GenerateWidgetFromWidgetAction(const AZStd::string& widgetActionIdentifier) const override;
```

Comment:
update: This member function should not be const if it's returning a non-const pointer (either have a const overload or remove the const)

Reply:
I agree on principle, but in this case `GenerateWidgetFromWidgetAction` is a function that creates a widget and releases it to the wild without keeping a reference to it. So technically it is const for the Action Manager.

I will make the change for consistency though.